pkg/asset/machines/openstack/machines.go

```diff
@@ -107,7 +108,70 @@ func Machines(clusterID string, config *types.InstallConfig, pool *types.Machine
 		machines = append(machines, machine)
 	}
 
-	return machines, nil
+	controlPlaneMachineSet := &machinev1.ControlPlaneMachineSet{}
+	replicas := int32(total)
+	failureDomains := []machinev1.OpenStackFailureDomain{}
+
+	if len(mpool.FailureDomains) > 0 {
```

Comment:
@pierreprinetti you'll notice how I consume both the TechPreview FailureDomain API that you introduced in 4.13 and the CPMS FailureDomain that we're working on now.

Reply:
@JoelSpeed @damdo
talked offline to @pierreprinetti and he wondered if we could call a function from CCPMSO here to build the `controlPlaneMachineSet` instead of having to implement a way here, that might be different from CPMS.

is that even possible? is it in the roadmap? or should we just do it like this for now.